src/nodeViews/RTENode.ts

```diff
@@ -0,0 +1,208 @@
+import { exampleSetup } from "prosemirror-example-setup";
+import { redo, undo } from "prosemirror-history";
+import { keymap } from "prosemirror-keymap";
+import type { Node, Schema } from "prosemirror-model";
+import type { Transaction } from "prosemirror-state";
+import { EditorState } from "prosemirror-state";
+import { Mapping, StepMap } from "prosemirror-transform";
+import type { Decoration } from "prosemirror-view";
+import { DecorationSet, EditorView } from "prosemirror-view";
+
+/**
+ * A NodeView (https://prosemirror.net/docs/ref/#view.NodeView) that represents a
+ * nested rich text editor interface.
+ */
+export class RTENodeView<LocalSchema extends Schema> {
+  // The parent DOM element for this view. Public
+  // so it can be mounted by consuming elements.
+  public dom = document.createElement("div");
+  // The editor view for this NodeView.
+  private innerEditorView: EditorView | undefined;
+  // The decorations that apply to this NodeView.
+  private decorations = new DecorationSet();
+
+  constructor(
+    // The node that this NodeView is responsible for rendering.
+    private node: Node,
+    // The outer editor instance. Updated from within this class when the inner state changes.
+    private outerView: EditorView,
+    // Returns the current position of the parent Nodeview in the document.
+    private getPos: boolean | (() => number),
```

Comment:
Might be worth outlining how this can be a boolean as I'm not totally clear on it.

Reply:
It's a great point. `getPos` is a Prosemirror function that has this signature, but I think we can make the interface clearer by making sure `getPos` is a function before we construct this nodeview, if that makes sense. So we should be able to wrap `getPos` and simplify this to just `() => number`. I'll make the change in a single commit in an attempt to make this clear.

Reply:
This was already the case (see `plugin.ts` L79), so that extra signature was redundant – I've removed it.